### PR TITLE
add eth_getCode method

### PR DESCRIPTION
**Pull request by bh2smith**
So far I have only added what I believe to be the interface. I am not sure if there are other parts of the code that need to be altered for this to work. Didn't see any tests I could work from. Please let me know if there is more to be done here.

cc @nlordell

### CODE REVIEW

File: crates/ethrpc/src/lib.rs [hunk 1]
```diff
@@ -49,6 +49,10 @@ module! {
         /// Returns a collection of all logs matching the given filter.
         pub struct GetLogs as "eth_getLogs"
             (LogFilter,) => Vec<Log>;
+
+        /// Returns code at a given address
+        pub struct GetCode as "eth_getCode"
+        (Address,) => Option<Vec<u8>>;
```
Comment (nlordell) [inline crates/ethrpc/src/lib.rs]: nit: can you indent this like the method above? Also, I think this needs a serialization directive (see the other method that returns a Vec<u8>)

**Review by nlordell — CHANGES_REQUESTED** [review]

File: crates/ethrpc/src/lib.rs (hunk 1 above)
Comment (bh2smith) [reply crates/ethrpc/src/lib.rs]: Right. I noticed the need for "serialization directive" but wasn't sure how it worked with the option there. Was waiting for you to say something about that.

**Comment by bh2smith** [comment]
It appears there is also an Optional Block number that can be provided to such a request. This is a feature I have never used, because I can't even imagine the reason for this.

one can specify: "latest", "earliest" or "pending". I suppose it wouldn't be so difficult to add here, but I would love to understand the difference first.

File: crates/ethrpc/src/lib.rs (hunk 1 above)
Comment (bh2smith) [reply crates/ethrpc/src/lib.rs]: Done

**Comment by bh2smith** [comment]
Ya, so as suspected the serializer doesn't like the Option:

```
the trait `AsRef<[u8]>` is not implemented for `std::option::Option<Vec<u8>>`
```

I can try to implement this.

Tried to do something like this:

```rs
impl AsRef<[u8]> for Option<Vec<u8>> {
    fn as_ref(&self) -> &[u8] {
        match self {
            Some(vec) => vec.as_ref(),
            None => &[],
        }
    }
}
```

in ethdigest, but it doesn't seem to like that either.

**Comment by bh2smith** [comment]
Turns out the Option was not necessary since the endpoint returns 0x when no code is available at the given address

File: crates/ethrpc/src/lib.rs [hunk 2]
```diff
@@ -49,6 +49,10 @@ module! {
         /// Returns a collection of all logs matching the given filter.
         pub struct GetLogs as "eth_getLogs"
             (LogFilter,) => Vec<Log>;
+
+        /// Returns code at a given address
```
Comment (nlordell) [inline crates/ethrpc/src/lib.rs]:
```suggestion
        /// Returns code at a given address.
```

**Review by nlordell — APPROVED** [review]

**Pull request merged**